src/parser/expression.js

```diff
@@ -1398,20 +1398,26 @@ export default class ExpressionParser extends LValParser {
   }
 
   parsePropertyName(
-    prop: N.ObjectOrClassMember | N.TsNamedTypeElementBase,
-  ): N.Expression {
+    prop: N.ObjectOrClassMember | N.ClassPrivateProperty | N.ClassPrivateMethod | N.TsNamedTypeElementBase,
+  ): N.Expression | N.Identifier {
     if (this.eat(tt.bracketL)) {
       prop.computed = true;
       prop.key = this.parseMaybeAssign();
       this.expect(tt.bracketR);
     } else {
-      prop.computed = false;
       const oldInPropertyName = this.state.inPropertyName;
       this.state.inPropertyName = true;
       prop.key =
         this.match(tt.num) || this.match(tt.string)
           ? this.parseExprAtom()
-          : this.parseIdentifier(true);
+          : this.parseMaybePrivateName();
+
+      // TODO: if no plugin and is PrivateName, should error be handled here or elsewhere?
+      if (prop.key.type !== "PrivateName") {
+        // ClassPrivateProperty is never computed, so we don't assign in that case.
+        prop.computed = false;
```

Comment:
`prop` here is a `MemberExpression`? In that case, it should always be set to `false`.

Reply:
I think the method signature says it's a `Class[Private]Property` or `Class[Private]Method`.